.travis.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+set -e
+
+show_help() {
+  echo "Usage: $0 <command>"
+  echo "Commands: install_docker"
+}
+
+if [[ -z $1 ]]; then
+  show_help
+  exit 1
+fi
+
+case "$1" in
+  install_docker)
+
+    if [[ -z $DOCKER_VERSION ]]; then
+      echo "DOCKER_VERSION needs to be set as an environment variable"
+      exit 1
+    fi
+
+    # TODO detect which docker version is already installed and skip
+    # uninstall/reinstall if it matches $DOCKER_VERSION
+
+    # stop docker service if running
+    sudo stop docker || :
+
+    sudo apt-get -qq update
+    sudo apt-get -q -y purge docker-engine
+    apt-cache policy docker-engine
+    sudo apt-get -q -y install docker-engine=$DOCKER_VERSION* "linux-image-extra-$(uname -r)"
+
+    if [[ "$DOCKER_VERSION" =~ ^1\.6\..* ]]; then
+      # docker-engine 1.6.x packages don't seem to have the upstart job config,
+      # so write it ourselves
+      echo '
+description "Docker daemon"
+start on (local-filesystems and net-device-up IFACE!=lo)
+stop on runlevel [!2345]
+limit nofile 524288 1048576
+limit nproc 524288 1048576
+respawn
+kill timeout 20
+script
+	DOCKER=/usr/bin/$UPSTART_JOB
+	DOCKER_OPTS="-D=true -H=unix:///var/run/docker.sock -H=tcp://127.0.0.1:2375"
+	exec "$DOCKER" -d $DOCKER_OPTS
+end script
+# Don'"'"'t emit "started" event until docker.sock is ready.
+# See https://github.com/docker/docker/issues/6647
+post-start script
+	DOCKER_OPTS="-D=true -H=unix:///var/run/docker.sock -H=tcp://127.0.0.1:2375"
+	while ! [ -e /var/run/docker.sock ]; do
+		initctl status $UPSTART_JOB | grep -qE "(stop|respawn)/" && exit 1
+		echo "Waiting for /var/run/docker.sock"
+		sleep 0.1
+	done
+	echo "/var/run/docker.sock is up"
+end script
```

Comment:
yea that probably would have been a lot more sensible than learning how to escape these single quotes!